airflow/providers/databricks/hooks/databricks_sql.py

```diff
@@ -248,7 +265,22 @@ def run(
                 self.set_autocommit(conn, autocommit)
 
                 with closing(conn.cursor()) as cur:
-                    self._run_command(cur, sql_statement, parameters)  # type: ignore[attr-defined]
+                    t = create_timeout_thread(cur, execution_timeout)
+
+                    # todo; adjust this to make testing easier
+                    try:
+                        self._run_command(cur, sql_statement, parameters)  # type: ignore[attr-defined]
+                    except Exception as e:
```

Comment:
Is it possible for us to carry a narrower exception?

Reply:
I don't believe it is possible since we want every error message to be propagated back; however, if it's due to the query being canceled by the thread `t`, we want to create a specific timeout error message instead of only propagating the error message from Databricks.

Reply:
When working on this, I found similar usage for other providers but am open to suggestions on how this could be improved: https://github.com/apache/airflow/blob/b0a18d9019a1a12d58435185b0c86e28af050685/airflow/providers/snowflake/operators/snowflake.py#L562-L573

Reply:
sounds good 👍